kaldb/src/main/java/com/slack/kaldb/chunkManager/ChunkManager.java

```diff
@@ -3,12 +3,14 @@
 import com.slack.kaldb.logstore.search.SearchQuery;
 import com.slack.kaldb.logstore.search.SearchResult;
 import com.slack.kaldb.metadata.schema.FieldType;
+import com.slack.service.murron.trace.Trace;
 import java.io.IOException;
 import java.time.Duration;
 import java.util.Map;
 
 public interface ChunkManager<T> {
-  void addMessage(T message, long msgSize, String kafkaPartitionId, long offset) throws IOException;
+  void addMessage(Trace.Span message, long msgSize, String kafkaPartitionId, long offset)
```

Comment:
so the goal is we will have field type info in the trace.proto and use that to create lucene documents. 

My motivation here was if we create lucene documents directly from trace.proto makes it much easier to preserve field type info. see [this](https://github.com/slackhq/kaldb/pull/778/files#diff-b88dc1d6c074deb00c77112f886e8e8993b793a04d7e665844d9b0977d1ddf8d) where I have a prototype that we'll get back to once this work is done

Reply:
keeping the interface templated is good, but then we'd need to change the query API to use the same object in one PR.

> I think it may be better to keep the LogMessage class but change it's implementation to be a Trace.Span so it's easier to make ingest format changes in the future

Do you have a concrete example of how this would look like? I'd love to see what you have in mind. Because the current way I am doing it means we need to change all the test cases which is not ideal.

Reply:
it is also of huge importance to use whatever object that the preprocessor creates be used to create lucene documents.

For example, today we have tests like `DropPolicyTest#testNestedDocumentCreation` that think we support nested fields. That is indeed correct that LogMessage supports nested fields

However today trace.span does not support nested fields. Which means in practice we've never executed that code path. But the fact we can write test cases with LogMessage allows for these scenarios which we want to avoid.


I 100% agree Trace.Span in the current form is not ideal, but that can keep evolving while maintaining back compat. having a translation layer just leads to more issues and doesn't give us any advantage.

Reply:
Thanks for a quick response. I see this as a one way change that would be very hard to undo as the project evolves. So, I need a day or so to write down my thoughts. I also opened a discussion https://github.com/slackhq/kaldb/discussions/791 since there are many PRs involved in this. 

I would appreciate if you can pause the any merges to master until that discussion concludes.

Reply:
I am curious about why this is important? Why can the pre-processor not produce multiple types?

> it is also of huge importance to use whatever object that the preprocessor creates be used to create lucene documents.

Reply:
> I also opened a discussion https://github.com/slackhq/kaldb/discussions/791 since there are many PRs involved in this.

sounds good! Let's move it there.